crates/turborepo-cache/src/fs.rs

```diff
@@ -41,14 +44,33 @@ impl FSCache {
     pub fn new(
         override_dir: Option<&Utf8Path>,
         repo_root: &AbsoluteSystemPath,
+        analytics_recorder: Option<AnalyticsSender>,
     ) -> Result<Self, CacheError> {
         let cache_directory = Self::resolve_cache_dir(repo_root, override_dir);
         cache_directory.create_dir_all()?;
 
-        Ok(FSCache { cache_directory })
+        Ok(FSCache {
+            cache_directory,
+            analytics_recorder,
+        })
     }
 
-    pub fn fetch(
+    async fn log_fetch(&self, event: analytics::CacheEvent, hash: &str, duration: u64) {
+        // If analytics fails to record, it's not worth failing the cache
+        if let Some(analytics_recorder) = &self.analytics_recorder {
+            let analytics_event = AnalyticsEvent {
+                session_id: None,
+                source: analytics::CacheSource::Local,
+                event,
+                hash: hash.to_string(),
+                duration,
+            };
+
+            let _ = analytics_recorder.send(analytics_event).await;
```

Comment:
Related to my comment about the unbounded channel, but I think we could get away with using `blocking_send` here to avoid holding a reference to `&self` across an await and we can drop the `Send` requirement on the reader. (If we don't swap to an unbounded receiver)

Reply:
I originally had it as `blocking_send` here, but since this is called in an async context (specifically in a tokio thread), it panics. Also, it seemed like a code smell to have code that panics if it's called in the wrong situation anyways